receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go

```diff
@@ -58,7 +57,16 @@ var Metrics = struct {
 		data.SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
 
 		return metric
-	}(),
+	},
+}
+
+const SystemCPUTime = "system.cpu.time"
+const SystemMemoryUsage = "system.memory.usage"
+
+// MetricsByName makes it easier to look up metrics at runtime by their name.
+var MetricsByName = map[string]func() pdata.Metric{
+	SystemCPUTime:     MetricFactories.SystemCPUTime,
+	SystemMemoryUsage: MetricFactories.SystemMemoryUsage,
 }
```

Comment:
Instead of the constants what about changing the existing Metrics struct to work like:

`Metrics.SystemCPUTime.String() or Name()` -> `"system.cpu.time"`
`Metrics.SystemCPUTime.Create()` -> `func() { return pdata.NewMetric(), etc. }`

Maybe eventually it'd have:

`Metrics.SystemCPUTime.Labels()` as well for the set of labels it can emit. 

This keeps things closer to how Labels are structured as well.

Maybe `MetricsByName` could map to the same structs, ie `MetricsByName["foo.bar"].Labels()`

Reply:
Yeah that seems like a good idea.

Reply:
@jrcamp alright did that except I called it `New` instead of `Create` since that is more consistent with Go constructors in general and pdata.